internal/testplanet/uplink.go

```diff
@@ -0,0 +1,261 @@
+	return pdbclient.NewClient(uplink.Identity, destination.Addr(), apikey)
+}
+
+// DialOverlay dials destination and returns an overlay.Client
+func (uplink *Uplink) DialOverlay(destination Peer) (overlay.Client, error) {
+	info := destination.Local()
+	conn, err := uplink.Transport.DialNode(context.Background(), &info, grpc.WithBlock())
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: handle disconnect
+	return overlay.NewClientFrom(pb.NewOverlayClient(conn)), nil
+}
+
+// Upload data to specific satellite
+func (uplink *Uplink) Upload(ctx context.Context, satellite *satellite.Peer, bucket string, path storj.Path, data []byte) error {
+	metainfo, streams, err := uplink.getMetainfo(satellite)
+	if err != nil {
+		return err
+	}
+
+	encScheme := uplink.getEncryptionScheme()
+	redScheme := uplink.getRedundancyScheme()
+
+	// create bucket if not exists
+	_, err = metainfo.GetBucket(ctx, bucket)
+	if err != nil {
+		if storj.ErrBucketNotFound.Has(err) {
+			_, err := metainfo.CreateBucket(ctx, bucket, &storj.Bucket{PathCipher: encScheme.Cipher})
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	createInfo := storj.CreateObject{
+		RedundancyScheme: redScheme,
+		EncryptionScheme: encScheme,
+	}
+	obj, err := metainfo.CreateObject(ctx, bucket, path, &createInfo)
+	if err != nil {
+		return err
+	}
+
+	reader := bytes.NewReader(data)
+	err = uploadStream(ctx, streams, obj, reader)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func uploadStream(ctx context.Context, streams streams.Store, mutableObject storj.MutableObject, reader io.Reader) error {
+	mutableStream, err := mutableObject.CreateStream(ctx)
+	if err != nil {
+		return err
+	}
+
+	upload := stream.NewUpload(ctx, mutableStream, streams)
+
+	_, err = io.Copy(upload, reader)
+
+	return errs.Combine(err, upload.Close())
+}
+
+// Download data from specific satellite
+func (uplink *Uplink) Download(ctx context.Context, satellite *satellite.Peer, bucket string, path storj.Path) ([]byte, error) {
+	metainfo, streams, err := uplink.getMetainfo(satellite)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	readOnlyStream, err := metainfo.GetObjectStream(ctx, bucket, path)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	download := stream.NewDownload(ctx, readOnlyStream, streams)
+	defer func() { err = errs.Combine(err, download.Close()) }()
+
+	buffer := bytes.NewBuffer([]byte{})
+	_, err = io.Copy(buffer, download)
+	if err != nil {
+		return []byte{}, err
+	}
+	return buffer.Bytes(), nil
+}
+
+func (uplink *Uplink) getMetainfo(satellite *satellite.Peer) (db storj.Metainfo, ss streams.Store, err error) {
+	encScheme := uplink.getEncryptionScheme()
+	redScheme := uplink.getRedundancyScheme()
+
+	// redundancy settings
+	minThreshold := int(redScheme.RequiredShares)
+	repairThreshold := int(redScheme.RepairShares)
+	successThreshold := int(redScheme.OptimalShares)
+	maxThreshold := int(redScheme.TotalShares)
+	erasureShareSize := 1 * memory.KB
```

Comment:
Use `KiB` and `MiB` instead of `KB` and `MB`.

Reply:
Now I have error message that `EncryptionBlockSize must be a multiple of ErasureShareSize * RS MinThreshold`

Reply:
Which means that we are using `KB` instead of `KiB` somewhere else. We did such a mess with these units.

Reply:
ok, I found it. it was in my code.